### PR TITLE
Feat/#55/notification feature

### PR DESCRIPTION
**Pull request by Zama15**
## Notification feature

Add update method to database model class.
Add where_conplex to database, due to the limits in the where method were added a new method who could cover more cases or need a complex query logic
Add triggers to a notification folder in the bd folder, this trigger are in charge to create the notification after any action made on the database
The notification are trigger in this cases:
- On follow
- On like
- On comments
- When a user you are following create a new post
The notification are mark as read in the moment the user click at the notification icon

## Fixed files

Modify confirmations_controller and passwords_controller file who have a update method were renamed to patch due to the recent added method in database model.
Add gitignore again on assets folder that were eliminated in lasts commits.
Fix create_comment procedure
Modify user model, find_by_id method because it didnt show the count of post the user had
Add the error 500 handler lines, previusly it have a bug were the error could be displayed properly 
Fix get_uri error on app.js file in app object, were in previus commits a join were added which lead in a error

### CODE REVIEW

File: models/user.php [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
+require_once "notification.php";
 
 class User extends Base {
   public function __construct() {
     try {
       $this->conn = $this->db_connection();
       $this->check_connection();
     } catch (Exception $e) {
-      throw new Exception("Failed to connect to the database: " . $e->getMessage());
+      // throw new Exception("Failed to connect to the database: " . $e->getMessage());
```
Comment (MeganPegueros) [inline models/user.php:12]: Delete line if commented

**Review by MeganPegueros — COMMENTED** [review]
Formatting correction

File: models/user.php (hunk 1 above)
Comment (Zama15) [reply models/user.php:12]: done

**Review by Gavalencia12 — COMMENTED** [review]
LGTM

**Review by AngelDiaz591 — APPROVED** [review]

**Pull request merged**